src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/VsSafeProjectGuidService.cs

```diff
@@ -13,18 +14,49 @@ internal class VsSafeProjectGuidService : ISafeProjectGuidService
 {
     private readonly UnconfiguredProject _project;
     private readonly IUnconfiguredProjectTasksService _tasksService;
+    private readonly IVsService<SVsBackgroundSolution, IVsBackgroundSolution> _backgroundSolutionImport;
```

Comment:
Can be simplified here. The service type is only needed on the import.

```suggestion
    private readonly IVsService<IVsBackgroundSolution> _backgroundSolutionImport;
```